phpunit_tests/RouterTest.php

```
@@ -0,0 +1,120 @@
+ * so it isn't directly testable in-process. These tests target the
+ * stateless / static helpers exposed on Router that the broader codebase
+ * (and Route, JsonData enum, handler tests) lean on.
+ */
+#[CoversClass(Router::class)]
+final class RouterTest extends TestCase
+{
+    /** @var array<string,mixed> */
+    private array $savedServer = [];
+
+    /** @var array<string,mixed> */
+    private array $savedEnv = [];
```

PHPMD (2.15.0)

[warning] Avoid unused private fields such as '$savedEnv'. (undefined)

(UnusedPrivateField)